src/sentry/scripts/digests/digests.lua

```diff
@@ -278,10 +278,15 @@ local function digest_timeline(configuration, timeline_id)
         if redis.call('EXISTS', digest_key) == 1 then
             -- If the digest set already exists (possibly because we already tried
             -- to send it and failed for some reason), merge any new data into it.
-            -- TODO: It might make sense to trim here to avoid returning capacity *
-            -- 2 if timeline was full when it was previously digested.
             redis.call('ZUNIONSTORE', digest_key, 2, timeline_key, digest_key, 'AGGREGATE', 'MAX')
             redis.call('DEL', timeline_key)
+
+            -- After merging, we have to do a capacity check (if we didn't,
+            -- it's possible that this digest could grow to an unbounded size
+            -- if it is never actually closed.)
+            if timeline_capacity > 0 then
```

Comment:
I didn't feel that it was necessary to do a probabilistic truncation here since this shouldn't happen as frequently as it does for record addition.